### DISCUSSION TITLE
Prisma Client Extension query Typesafety

### DISCUSSION THREAD

**Discussion by mshndev** [prisma/prisma#21663] [Q&A]
### Question

I'm trying to create a Prisma Client Extension that can add specific imports to a model.

```typescript
const postExtention = Prisma.defineExtension((prisma) =>
  prisma.$extends({
    name: 'postExtention',
    query: {
      post: {
        $allOperations({ operation, args, query }) {
          if (
            operation === 'createMany' ||
            operation === 'updateMany' ||
            operation === 'deleteMany' ||
            operation === 'aggregate' ||
            operation === 'count' ||
            operation === 'groupBy'
          )
            return query(args)

          return query({
            ...args,
            // should include a User relation in return type
            include: {
              ...(args.include || {}),
              user: true
            }
          })
        }
      }
    }
  })
)

// Usage
function getPost (id: string) {
  return prisma.$extends(postExtention).post.findUnique({ where: { id } })
}

// NO user in PostOutput
type PostOutput = Awaited<ReturnType<typeof getPost>>
```

The problem is that the `PostOutput` type being a simple `Prisma.Post` type, lacking the `user` relation.

I experimented with using specific methods within the extension, such as `findUnique` instead of `$allOperations`, but this didn't resolve the issue as expected.

The only way I found is to modify a generic of the `findUnique` method:

```typescript
prisma.$extends(postExtention).post.findUnique<{where: {id: string}, include: {user: boolean}}>({ where: { id } })
```

But its not a solution for me.

How can I modify the extension to ensure that it considers all the includes I specify in the result type, without adding to the generic of `findUnique`? I would greatly appreciate any guidance on this matter.

### Information about Prisma Schema, Client Queries and Environment (optional)

Prisma version is `5.4.1`

**Comment by ludralph** [comment]
Hi @sergeymishin 

I'm sharing this with the team. We will get back to you with a response soon.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
